infrastructures/docker/infrastructure/src/main/java/org/eclipse/che/workspace/infrastructure/docker/DockerInfraModule.java

```diff
@@ -67,5 +70,11 @@ protected void configure() {
     bind(
         org.eclipse.che.workspace.infrastructure.docker.monit.DockerAbandonedResourcesCleaner
             .class);
+
+    if (Boolean.parseBoolean(System.getenv("CHE_SINGLE_PORT"))) {
```

Comment:
I would suggest not to use conditional binding. We are usually complaining such solutions. Instead of that you can use technique that was used in Che 5 https://github.com/eclipse/che/blob/5.22.x/dockerfiles/init/modules/che/templates/che.env.erb#L81